cdc/scheduler/balancer.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
+
+import (
+	"math"
+
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/cdc/scheduler/util"
+	"go.uber.org/zap"
+)
+
+// balancer is used to support the rebalance process, in which
+// some victims are chosen and de-scheduled. Later, the victims
+// will be automatically rescheduled, during which the target captures
+// will be chosen so that the workload is the most balanced.
+type balancer interface {
+	// FindVictims returns a set of possible victim tables.
+	// Removing these tables will make the workload more balanced.
+	FindVictims(
+		tables *util.TableSet,
+		captures map[model.CaptureID]*model.CaptureInfo,
+	) (tablesToRemove []model.TableID)
```

Comment:
why not return map[CaptureID][tableIDList]

Reply:
We can return `[]TableRecord`. In the current design, we look up the `TableSet` to know which capture these tables are assigned to.